clients/python/samsara-sdk/client.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+"""
+    samsara-sdk.client
+    ~~~~~~~~~~~~~~~~~~
+
+    Samsara SDK client
+"""
+
+from cerberus import Validator
+import gzip
+import json
+import time
+import requests
+
+
+class InvalidEvent(Exception):
+    pass
+
+
+class InvalidArguement(Exception):
+    pass
+
+
+class COMPRESSION(object):
+    GZIP = "gzip"
+
+    @staticmethod
+    def compress_none(s):
+        return json.dumps(s)
+
+    @staticmethod
+    def compress_gzip(s):
+        return gzip.compress(bytes(json.dumps(s), 'utf-8'))
+
+    COMPRESS = {
+        GZIP: compress_gzip,
+        None: compress_none
+    }
+
+    @staticmethod
+    def get_compressor(compression):
+        if compression not in COMPRESSION.COMPRESS.keys():
+            raise InvalidArguement(
+                "Compression has to be one of {}"
+                .format(COMPRESSION.COMPRESS.keys()))
+
+        return COMPRESSION.COMPRESS.get(compression)
+
+
+PUBLISHED_TIMESTAMP = "X-Samsara-publishedTimestamp"
+DEFAULT_CONFIG = {
+    # a samsara ingestion api endpoint  "http://samsara.io/"
+    # url  - REQUIRED
+
+    # the identifier of the source of these events
+    # source_id  - OPTIONAL only for record-event
+
+    # whether to start the publishing thread.
+    "start_publishing_thread": True,
+
+    # how often should the events being sent to samsara
+    # in milliseconds
+    # default 30s
+    "publish_interval": 30000,
+
+    # max size of the buffer, when buffer is full
+    # older events are dropped.
+    "max_buffer_size": 10000,
+
+    # minimum number of events to that must be in the buffer
+    # before attempting to publish them
+    "min_buffer_size": 100,
+
+    # network timeout for send operations (in seconds)
+    # default 30s
+    "send_timeout": 30,
+
+    # whether of not the payload should be compressed
+    # allowed values "gzip" "none"
+    "compression": COMPRESSION.GZIP
+
+    # add samsara client statistics events
+    # this helps you to understand whether the
+    # buffer size and publish-intervals are
+    # adequately configured.
+    # "send_client_stats": True
+}
+
+
+event_validator = Validator(
+    schema={
+        'source_id': {'required': True, 'type': 'string'},
+        'timestamp': {'required': True, 'type': 'integer', 'min': 0},
+        'event_name': {'required': True, 'type': 'string'}
+    }, allow_unknown=True)
```

Comment:
Here fields must be `sourceId` and `eventName` exactly, otherwise the `ingestion-api` will reject the request.